samples/unit-tests/legend/legend-general/demo.js

```diff
@@ -130,6 +130,18 @@ QUnit.test('Legend redraws', function (assert) {
     chart.series[0].hide(true);
 
     assert.strictEqual(visible, false, 'Legend item text has changed (#2165)');
+
+    chart.series[0].remove();
+
+    chart.update({
+        legend: {
+            title: {
+                text: 'Legend'
+            }
+        }
+    });
+
+    assert.strictEqual(chart.legend.group.visibility === 'hidden', true, 'Legend title is not visible (#17167)');
```

Comment:
You have 2 checks here: `visibility === 'hidden'` and `true === true` (`strictEqual`);
Try to find a way how to get rid of `===` to keep one check only (`strictEqual`).